endpoints/osp/osp

```diff
@@ -0,0 +1,582 @@
+#!/usr/bin/env bash
+# -*- mode: sh; indent-tabs-mode: nil; sh-basic-offset: 4 -*-
+# vim: autoindent tabstop=4 shiftwidth=4 expandtab softtabstop=4 filetype=bash
+
+# This script implements the 'osp' endpoint for rickshaw.  It runs 1 or more
+# clients and servers as instances/VMs in an Openstack cluster.
+#
+# Usage:
+# osp [--validate] --endpoint-opts=host=<host>,user=<user>,client:n-m,o-p,server:n-m,o-p 
+#                    --run-id <id> --base-run-dir --image <location>
+#                    --roadblock-server <host> --roadblock-id <id> --roadblock-passwd=<passwd>
+#
+# If --validate is used all options after client/server will be ignored
+#
+
+# Source the base file for common functions and config
+this_endpoint_dir=$(dirname `readlink -e $0` | sed -e 'sX/binXX')
+endpoint_base_dir=$(cd $this_endpoint_dir >/dev/null && cd .. && /bin/pwd)
+if [ -e "$endpoint_base_dir/base" ]; then
+    . "$endpoint_base_dir/base"
+else
+    echo "Could not find endpoint source file "$endpoint_base_dir/base", exiting"
+    exit 1
+fi
+
+echo "#running osp endpoint"
+clear_all_servers="1"
+chroot_rbind_mounts="proc dev sys lib/modules usr/src boot var/run"
+image_cache_size=3
+default_flavor="nfv-large"
```

Comment:
this seems like an odd default

Reply:
Sorry, I meant the `default_flavor`

Reply:
Ah, yes, I should probably change that.  However, I am not sure what value to use.  Perhaps we set it to "default", which IMO is totally useless, but then when it checks for the flavor and errors out, the user will then know they need to provide the flavor.

Reply:
No idea if this is representative or not, but in my limited work with `microstack` so far I'm finding `m1.small` to be the minimum required flavor (`m1.tiny` does not create a disk big enough for the the default Centos Stream 9 and Ubuntu 22.04 images I downloaded).